src/test/java/com/commercetools/sync/shoppinglists/ShoppingListSyncOptionsBuilderTest.java

```diff
@@ -0,0 +1,273 @@
+            .batchSize(30)
+            .beforeCreateCallback((newShoppingList) -> null)
+            .beforeUpdateCallback((updateActions, newShoppingList, oldShoppingList) -> emptyList())
+            .build();
+        assertThat(shoppingListSyncOptions).isNotNull();
+    }
+
+    @Test
+    void batchSize_WithPositiveValue_ShouldSetBatchSize() {
+        final ShoppingListSyncOptions shoppingListSyncOptions = ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
+                                                                                              .batchSize(10)
+                                                                                              .build();
+        assertThat(shoppingListSyncOptions.getBatchSize()).isEqualTo(10);
+    }
+
+    @Test
+    void batchSize_WithZeroOrNegativeValue_ShouldFallBackToDefaultValue() {
+        final ShoppingListSyncOptions shoppingListSyncOptionsWithZeroBatchSize =
+            ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
+                                          .batchSize(0)
+                                          .build();
+
+        assertThat(shoppingListSyncOptionsWithZeroBatchSize.getBatchSize())
+            .isEqualTo(ShoppingListSyncOptionsBuilder.BATCH_SIZE_DEFAULT);
+
+        final ShoppingListSyncOptions shoppingListSyncOptionsWithNegativeBatchSize = ShoppingListSyncOptionsBuilder
+            .of(CTP_CLIENT)
+            .batchSize(-100)
+            .build();
+        assertThat(shoppingListSyncOptionsWithNegativeBatchSize.getBatchSize())
+            .isEqualTo(ShoppingListSyncOptionsBuilder.BATCH_SIZE_DEFAULT);
+    }
+
+    @Test
+    void applyBeforeUpdateCallBack_WithNullCallback_ShouldReturnIdenticalList() {
+        final ShoppingListSyncOptions shoppingListSyncOptions = ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
+                                                                                              .build();
+        assertThat(shoppingListSyncOptions.getBeforeUpdateCallback()).isNull();
+
+        final List<UpdateAction<ShoppingList>> updateActions = singletonList(ChangeName.of(ofEnglish("name")));
+
+        final List<UpdateAction<ShoppingList>> filteredList =
+            shoppingListSyncOptions.applyBeforeUpdateCallback(
+                updateActions, mock(ShoppingListDraft.class), mock(ShoppingList.class));
+
+        assertThat(filteredList).isSameAs(updateActions);
+    }
+
+    @Test
+    void applyBeforeUpdateCallBack_WithNullReturnCallback_ShouldReturnEmptyList() {
+        final TriFunction<List<UpdateAction<ShoppingList>>, ShoppingListDraft,
+            ShoppingList, List<UpdateAction<ShoppingList>>> beforeUpdateCallback =
+                (updateActions, newShoppingList, oldShoppingList) -> null;
+
+        final ShoppingListSyncOptions shoppingListSyncOptions = ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
+                                                                                              .beforeUpdateCallback(
```

Comment:
The formatting of those snippets could be better.

Reply:
like? i used the code template you provided to format code.

Reply:
the code snippet below hard to read, so it could be better.

for example:

````java
final ShoppingListSyncOptions shoppingListSyncOptions = ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
                                                                                              .beforeUpdateCallback(
                                                                                          beforeUpdateCallback)
                                                                                              .build();
````

to

```` java
final ShoppingListSyncOptions shoppingListSyncOptions = 
            ShoppingListSyncOptionsBuilder.of(CTP_CLIENT)
                                          .beforeUpdateCallback(beforeUpdateCallback)
                                          .build();
````

Reply:
01fe21a

Reply:
I think you can change this two line as well
https://github.com/commercetools/commercetools-sync-java/blob/01fe21a1b0fe024fdcda2aec697b05abfce79f4c/src/test/java/com/commercetools/sync/shoppinglists/ShoppingListSyncOptionsBuilderTest.java#L167
https://github.com/commercetools/commercetools-sync-java/blob/01fe21a1b0fe024fdcda2aec697b05abfce79f4c/src/test/java/com/commercetools/sync/shoppinglists/ShoppingListSyncOptionsBuilderTest.java#L212

Reply:
I fixed in my branch and so i cherrypicked it to here
https://github.com/commercetools/commercetools-sync-java/pull/595/commits/91571fac6f627456261b9ac77bbc6c692f130278